dask/dataframe/shuffle.py

```diff
@@ -34,11 +34,19 @@ def set_index(
     if isinstance(index, Series) and index._name == df.index._name:
         return df
     if isinstance(index, (DataFrame, tuple, list)):
-        raise NotImplementedError(
-            "Dask dataframe does not yet support multi-indexes.\n"
-            "You tried to index with this index: %s\n"
-            "Indexes must be single columns only." % str(index)
-        )
+        # Accept ["a"], but not [["a"]]
+        if (
+            isinstance(index, (tuple, list))
+            and len(index) == 1
+            and not isinstance(index[0], (list, tuple))
```

Comment:
```suggestion
```

Reply:
This check was to avoid unpacking `index = [["a"]]` into `index=["a"]`.
Otherwise, the `NotImplementedError` will still raise, but will possibly confuse the user:
`You tried to index with this index: ["a"]`, when `[["a"]]` is what the user supplied.